material-overrides/main.html

```diff
@@ -28,6 +28,14 @@
   {%- endif -%}
 {%- endblock -%}
 
+{% block libs %}
+  {{ super() }}
+  <script>
+    // Anti-flicker
+    document.documentElement.classList.add('js-loading');
```

Comment:
The entire script should be placed in the `libs` block so it runs immediately, even before the DOM is fully parsed.